### ISSUE TITLE
Archer NPCs can damage ignored NPCs

### ISSUE THREAD

**Issue by Blamo27** [mcmonkey4eva/Sentinel#202]
Hello,

Even with "protected" config option set to true, archer NPCs can damage other NPCs even when they are supposed to ignore them : let's say, an archer ignores NPCs but attacks players, it fires an arrow to a player but a NPC is in the way, if the NPC takes the arrow it will be damaged by this arrow. That means that the "protected" option only prevents PLAYERS from damaging sentinels who ignore them, it doesn't prevent sentinels from damaging other sentinels (although they are not supposed to be able to harm them).

This is an issue with questing plugins : if you have to kill an NPC, and that NPC dies in the battle by the arrows of another NPC, the kill objective won't be completed by the player, who won't notice and continue on without knowing that he is still expected to kill that NPC.

Therefore if you could extend the "protected" config option so it works with sentinels between them too, we would greatly appreciate, thank you.

**Comment by mcmonkey4eva** [comment]
You probably want the `safeshot` option on the attacking NPC.
Alternately, tell the receiving NPC to ignore the attacking NPC.

**Comment by Ferocimo** [comment]
(sorry for the delay we were working on other things)

We tried all of that but that didn't work. Whatever we do, archers will still deal the damage to the other NPC (even with safeshot enabled AND ignoring NPCs in general and the NPC's name).

Like we said in the original post, the "protected" option doesn't seem to work with NPCs, only with players. And the safeshot option doesn't work either in such a case.

**Comment by Blamo27** [comment]
@mcmonkey4eva Any news ?

**Comment by Ferocimo** [comment]
Bumping this, we're still having this issue !

**Comment by mcmonkey4eva** [comment]
Try build DEV-160+ ( http://ci.citizensnpcs.co/job/Sentinel/ )

Sorry for the slowness in fixing this.

As far as I could tell, everything was working fine and you shouldn't have had that issue.
And then I remembered today that you might be one of the servers that uses a horribly non-standard config.
And I realized that the `enforce damage` option would have pushed its way through all protection/safety checks.
And I'm assuming/hoping that's what went wrong here.
If the newest build doesn't work out, I'm going to have to figure out how to gather a lot more information :P
